### ISSUE TITLE
No rules found within configuration.

### ISSUE THREAD

**Issue by Dryymoon** [tomasAlabes/stylefmt-loader#3]
Error: No rules found within configuration. Have you provided a "rules" property?
[0]     at exports.default (/SHARA/domik-frontend/node_modules/stylelint/dist/utils/configurationError.js:8:13)

```javascript
config.module.loaders.push({
  test: /\.scss$/,
  loader: ExtractTextPlugin.extract({
    fallbackLoader: 'style',
    loader: ((prod || devssr) ? 'overrider!' : '') +
    'css?importLoaders=1&localIdentName=[local]' + (!devcli && '&sourceMap=true') + // &localIdentName=[local]
//    '!svg-fill/encodeSharp' +
    '!postcss' +
    '!sass?outputStyle=expanded&sourceMap=true&sourceMapContents=true' +
    '!stylefmt?config=.stylelintrc'
  })
});
```

I`m trying to set any possible variants to config string, and without config, but no luck.

**Comment by tomasAlabes** [comment]
`stylelint` is triggering the exception, not `stylefmt`.
I don't have experience with the `ExtractTextPlugin`, try isolating the problem and/or trying it by pieces. Check the path to stylelintrc.
Sorry, I can't do much with that info...

**Comment by stewartyu** [comment]
I'm pretty sure this is a problem with the loader actually - I'm running into the same issue. I opened up `node_modules/stylelint/dist/augmentConfig.js` and added a console.log statement for `augmentedConfig`.

The first few errors shown are from `stylefmt-loader` where `augmentedConfig` is a string, which is why it can't find any rules. Further down, it's running `stylelint` outside of `stylefmt-loader`, and you can see that it is an object, so it works properly. I'm thinking that `stylefmt-loader` is passing a string of the file path to the config instead of an object.

**Comment by tomasAlabes** [comment]
Oh ok, so in 5.x that changed. Thanks for looking into it!
I will merge your PR and update the loader accordingly. But first I want to fix the other issue as I think it affects both versions (4.x and 5.x).
